src/java/eu/rssw/rcode/ClassCompilationUnit.java

```diff
@@ -0,0 +1,217 @@
+/**
+ * Copyright 2011-2016 Riverside Software
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ */
+package eu.rssw.rcode;
+
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.StringTokenizer;
+
+import javax.xml.bind.JAXBContext;
+import javax.xml.bind.JAXBException;
+import javax.xml.bind.Marshaller;
+import javax.xml.bind.annotation.XmlAttribute;
+import javax.xml.bind.annotation.XmlElement;
+import javax.xml.bind.annotation.XmlRootElement;
+
+import com.openedge.core.reflect.EventElement;
+import com.openedge.core.reflect.MethodElement;
+import com.openedge.core.reflect.PropertyElement;
+import com.openedge.core.reflect.RCodeClassInfo;
+import com.openedge.core.reflect.RCodeReadException;
+import com.openedge.core.reflect.TypeInformation;
+
+@XmlRootElement(name = "unit", namespace = "")
+public class ClassCompilationUnit {
+    @XmlAttribute
+    public String packageName, className, inherits;
+    @XmlElement
+    public List<String> interfaces = new ArrayList<>();
+    @XmlAttribute
+    public boolean isAbstract, isFinal, isInterface, isEnum;
+
+    @XmlElement(name = "classComment")
+    public List<String> classComment = new ArrayList<>();
+    @XmlElement(name = "constructor")
+    public List<Constructor> constructors = new ArrayList<>();
+    @XmlElement(name = "method")
+    public List<Method> methods = new ArrayList<>();
+    @XmlElement(name = "property")
+    public List<Property> properties = new ArrayList<>();
+    @XmlElement(name = "event")
+    public List<Event> events = new ArrayList<>();
+    @XmlElement(name = "using")
+    public List<Using> usings = new ArrayList<>();
+    @XmlElement(name = "member")
+    public List<EnumMember> enumMembers = new ArrayList<>();
+
+    public void classToXML(File out) throws JAXBException, IOException {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Refactor this method to throw at most one checked exception instead of: javax.xml.bind.JAXBException, java.io.IOException [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://sonar.riverside-software.fr/coding_rules#rule_key=squid%3AS1160)